plugins/ci/skills/payload-snapshot/scripts/payload_snapshot.py

```
@@ -410,41 +467,140 @@ def _all_blocking_passed(self, details: dict) -> bool:
         )
 
     def build(self, start_tag: str) -> list[str]:
-        chain = [start_tag]
+        tag_names = [
+            tag["release_tag"] for tag in self.sippy.fetch_tags()
+            if tag.get("release_tag")
+        ]
+        try:
+            start_index = tag_names.index(start_tag)
+        except ValueError:
+            raise ValueError(f"Tag {start_tag} not found in Sippy")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)